_user_guides/armory-pipeline-templates.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: post
+title: Armory Pipeline Templates (Dinghy)
+order: 106
+published: True
+---
+
+Armory Pipeline Templates provide a way of specifying pipeline definitions in source code repos (like GitHub & BitBucket). 
+
+The Armory Spinnaker installation provides a service called "Dinghy" which will keep the pipeline in Spinnaker in sync with what is defined in the GitHub repo. Also, users will be able to make a pipeline by composing other pipelines, stages, or tasks and templating certain values.
+
+## Primitives
+
+- **Modules**: These are templates that define a Stage/Task in the pipeline. They are kept in a single GitHub repo that is configurable when the dinghy service starts. eg:
+
+![dinghy-templates](http://f.cl.ly/items/3R0B3W3o2l3h2K0E3e1G/dinghy-template-repo.png)
+
+ They are JSON files with replacable values in them. e.g., a module that defines a wait stage in a pipeline might look like:
+```
+{
+    "name": "Wait",
+    "refId": "1",
+    "requisiteStageRefIds": [],
+    "type": "wait",
+    "waitTime": 42
+}
+```
+- **Pipeline definitions**: These define a pipeline for an application in a file called `dinghyfile`. The `dinghyfile` usually resides at the root level of the application repo. eg:
+
+![dinghyfile](http://f.cl.ly/items/3t3z0Q2Z040f0i0V2P3O/dinghyfile.png)
+
+You can compose stage/task templates to make a full definition. e.g., a Pipeline definition for a spinnaker application called `foo` that has a single wait stage might look like:
+```
+{
+  "application": "foo",
+  "pipelines": [
+    {
+      "application": "foo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {
+          "name": "Wait",
+          "refId": "1",
+          "requisiteStageRefIds": [],
+          "type": "wait",
+          "waitTime": 10
+        }
+      ],
+      "triggers": []
+    }
+  ]
+}
+```
+
+## Template variables and substitution
+
+We can have Pipeline definitions use Modules defined in another GitHub Repo. e.g.:
+```{% raw %}
+{
+  "application": "foo",
+  "pipelines": [
+    {
+      "application": "foo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{ module "wait.stage.module" }}
```

Comment:
it would be kind of cool if you could do 
```
{{ module "wait.stage.module" }}    // pipeline module created in dinghy-templates repo
```

but that's not valid json. soo.... up to you lol